src/xdocs/config_filters.xml

```diff
@@ -990,9 +1006,134 @@ public class InputTest {
+            @Generated("first") // should not be suppressed
+            public void test1() {
+            }
+
+            @Generated("second") // should be suppressed
+            public void test2() {
+            }
+        }
+        </source>
+        <p>First of all we need to look at AST tree printed by our CLI tool: </p>
+        <source>
+$ java -jar checkstyle-X.XX-all.jar -t InputTest.java
+CLASS_DEF -> CLASS_DEF [1:0]
+|--MODIFIERS -> MODIFIERS [1:0]
+|   `--LITERAL_PUBLIC -> public [1:0]
+|--LITERAL_CLASS -> class [1:7]
+|--IDENT -> InputTest [1:13]
+`--OBJBLOCK -> OBJBLOCK [1:23]
+|--LCURLY -> { [1:23]
+|--METHOD_DEF -> METHOD_DEF [2:4]
+|   |--MODIFIERS -> MODIFIERS [2:4]
+|   |   |--ANNOTATION -> ANNOTATION [2:4]
+|   |   |   |--AT -> @ [2:4]
+|   |   |   |--IDENT -> Generated [2:5]
+|   |   |   |--LPAREN -> ( [2:14]
+|   |   |   |--EXPR -> EXPR [2:15]
+|   |   |   |   `--STRING_LITERAL -> "first" [2:15]
+|   |   |   `--RPAREN -> ) [2:22]
+|   |   `--LITERAL_PUBLIC -> public [3:4]
+|   |--TYPE -> TYPE [3:11]
+|   |   `--LITERAL_VOID -> void [3:11]
+|   |--IDENT -> test1 [3:16]
+|   |--LPAREN -> ( [3:21]
+|   |--PARAMETERS -> PARAMETERS [3:22]
+|   |--RPAREN -> ) [3:22]
+|   `--SLIST -> { [3:24]
+|       `--RCURLY -> } [4:4]
+|--METHOD_DEF -> METHOD_DEF [6:4]
+|   |--MODIFIERS -> MODIFIERS [6:4]
+|   |   |--ANNOTATION -> ANNOTATION [6:4]
+|   |   |   |--AT -> @ [6:4]
+|   |   |   |--IDENT -> Generated [6:5]
+|   |   |   |--LPAREN -> ( [6:14]
+|   |   |   |--EXPR -> EXPR [6:15]
+|   |   |   |   `--STRING_LITERAL -> "second" [6:15]
+|   |   |   `--RPAREN -> ) [6:23]
+|   |   `--LITERAL_PUBLIC -> public [7:4]
+|   |--TYPE -> TYPE [7:11]
+|   |   `--LITERAL_VOID -> void [7:11]
+|   |--IDENT -> test2 [7:16]
+|   |--LPAREN -> ( [7:21]
+|   |--PARAMETERS -> PARAMETERS [7:22]
+|   |--RPAREN -> ) [7:22]
+|   `--SLIST -> { [7:24]
+|       `--RCURLY -> } [8:4]
+`--RCURLY -> } [9:0]
+        </source>
+        <p>AST node <code>ANNOTATION -> ANNOTATION [6:4]</code> has direct child
+        <code>IDENT -> Generated [6:5]</code>, therefore can be queried by <code>IDENT</code> value:
+        </p>
+        <source>
+&lt;suppress-xpath checks=&quot;.&quot; query=&quot;//METHOD_DEF[
+            .//ANNOTATION/IDENT[@text='Generated']]&quot;/&gt;
+        </source>
+        <p>The problem with query above that it will suppress violations for all methods with
+        annotation <code>@Generated</code>. In order to suppress methods with
+        <code>@Generated("second")</code> annotations only, you need to look at AST tree again.
+        Value of the <code>ANNOTATION</code> node is stored inside sub-node with token type
+        <code>STRING_LITERAL</code>. Use the following query to suppress methods with
+        <code>@Generated("second")</code> annotation:</p>
+        <source>
+&lt;suppress-xpath checks=&quot;.&quot; query=&quot;//METHOD_DEF[.//ANNOTATION[
+            ./IDENT[@text='Generated'] and .//*[@text='second']]]&quot;/&gt;
```

Comment:
> `//*`

please update example and put in Generated annotation a comment with "second" value too, https://github.com/javaee/javax.annotation/blob/master/src/main/java/javax/annotation/Generated.java, so xpath should be more strict and points only to value (not a comment) element of annotation

Reply:
it is not a comment, `.//*` means any subchild of the current node, I rewrote query, made it more strict